cub/cub/device/device_segmented_reduce.cuh

```diff
@@ -1131,23 +1178,21 @@ struct DeviceSegmentedReduce
     // TODO Address https://github.com/NVIDIA/cub/issues/651
     InitT initial_value{AccumT(1, Traits<InputValueT>::Lowest())};
 
-    return DispatchSegmentedReduce<ArgIndexInputIteratorT,
-                                   OutputIteratorT,
-                                   BeginOffsetIteratorT,
-                                   EndOffsetIteratorT,
-                                   OffsetT,
-                                   cub::ArgMax,
-                                   InitT,
-                                   AccumT>::Dispatch(d_temp_storage,
-                                                     temp_storage_bytes,
-                                                     d_indexed_in,
-                                                     d_out,
-                                                     num_segments,
-                                                     d_begin_offsets,
-                                                     d_end_offsets,
-                                                     cub::ArgMax(),
-                                                     initial_value,
-                                                     stream);
+    using integral_offset_check = ::cuda::std::is_integral<OffsetT>;
+    static_assert(integral_offset_check::value, "Offset iterator value type should be integral.");
+
+    return segmented_reduce<ArgIndexInputIteratorT, OutputIteratorT, BeginOffsetIteratorT, EndOffsetIteratorT, OffsetT>(
```

Comment:
important: we used to explicitly specify `AccumT`. Now, there's a logic that relies on return type of reduction operator. To be on the safe side, let's pass all of the template arguments as we used to do: reduction op, init value type, accumulator type. Same in other places.